config/main.go

```diff
@@ -0,0 +1,235 @@
+/*
+Copyright © 2024 Josh Holbrook <josh.holbrook@gmail.com>
+*/
+package config
+
+import (
+	_ "embed"
+	"errors"
+	"os"
+	"os/exec"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	"github.com/joho/godotenv"
+	"github.com/rs/zerolog/log"
+
+	"github.com/jfhbrook/dosapp/task"
+)
+
+//go:embed dosapp.env
+var envFile []byte
+
+//go:embed Taskfile.yml
+var taskFile []byte
+
+type Config struct {
+	// TODO: Root is only necessary until I move templating into dosapp
+	Root         string
+	ConfigHome   string
+	LogLevel     string
+	DosBoxBin    string
+	SevenZipBin  string
+	DataHome     string
+	StateHome    string
+	CacheHome    string
+	DiskHome     string
+	LinkHome     string
+	PackageHome  string
+	DownloadHome string
+	Pager        string
+	DiskA        string
+	DiskB        string
+	DiskC        string
+}
+
+func getEnv(key string, fallback string) string {
+	if value := os.Getenv(key); value != "" {
+		return value
+	}
+	return fallback
+}
+
+func expandUser(path string) (string, error) {
```

Comment:
The alternative to this is sourcing and printing the env, which might be preferable honestly. But this is OK for now.